src/MiniJuvix/Syntax/Abstract/Language/Extra.hs

```diff
@@ -26,12 +26,9 @@ smallerPatternVariables = \case
 
 viewApp :: Expression -> (Expression, [Expression])
 viewApp e = case e of
-  ExpressionIden {} -> (e, [])
   ExpressionApplication (Application l r) ->
     second (`snoc` r) (viewApp l)
-  ExpressionUniverse {} -> (e, [])
-  ExpressionFunction {} -> (e, [])
-  ExpressionLiteral {} -> (e, [])
+  _ -> (e, [])
```

Comment:
Revert this change. We list all the patterns.

Reply:
I agree that the rule of thumb should be to always list all the patterns, but in this case we are unfolding an application. There is only one constructor that we need to consider. Moreover, even if we add more constructors to Expression, the only relevant constructor will remain Application